AirCasting/CoreData/UIStorage.swift

```diff
@@ -0,0 +1,69 @@
+// Created by Lunar on 10/02/2022.
+//
+
+import CoreData
+import Combine
+import SwiftUI
+import Resolver
+
+protocol UIStorage {
+    func accessStorage(_ task: @escaping(HiddenCoreDataUIStorage) -> Void)
+}
+
+protocol UIStorageContextUpdate {
+    func cardStateToggle(for sessionUUID: SessionUUID) throws
+    func changeStream(for sessionUUID: SessionUUID, stream: String) throws
+    func save() throws
+}
+
+final class CoreDataUIStorage: UIStorage {
+    
+    @Injected private var persistenceController: PersistenceController
+    private lazy var updateSessionParamsService = UpdateSessionParamsService()
+    private lazy var context: NSManagedObjectContext = persistenceController.editContext
+    private lazy var hiddenStorage = HiddenCoreDataUIStorage(context: self.context)
+    
+    /// All actions performed on CoreDataMeasurementStreamStorage must be performed
+    /// within a block passed to this methood.
+    /// This ensures thread-safety by dispatching all calls to the queue owned by the NSManagedObjectContext.
+    func accessStorage(_ task: @escaping(HiddenCoreDataUIStorage) -> Void) {
+        context.perform {
+            task(self.hiddenStorage)
+            try? self.hiddenStorage.save()
+        }
+    }
+}
+
+final class HiddenCoreDataUIStorage: UIStorageContextUpdate {
+    
+    private let context: NSManagedObjectContext
+    
+    init(context: NSManagedObjectContext) {
+        self.context = context
+    }
+    
+    func cardStateToggle(for sessionUUID: SessionUUID) throws {
+        let sessionEntity = try context.existingSession(uuid: sessionUUID)
+        checkUIStateAvaibility(entity: sessionEntity)
+        sessionEntity.userInterface?.expandedCard.toggle()
+        try context.save()
+    }
+    
+    func changeStream(for sessionUUID: SessionUUID, stream: String) throws {
+        let sessionEntity = try context.existingSession(uuid: sessionUUID)
+        checkUIStateAvaibility(entity: sessionEntity)
```

Comment:
I'm thinking if there is a name that better describes the behavior of this method. As I understand it creates the UIState entity for session if it didn't have it yet, so maybe it could be named: `createUIStateForSessionIfNeeded` or something like that?

Reply:
I need to say your idea sound better than mine. Thank you 😄 I will change in a moment